### PR TITLE
create endpoint to add participant to database (fixes #59)

### LINKED ISSUE #59: Create endpoint to add participant to database

### PR DESCRIPTION
**Pull request by w1stler**
How to test:

- run postman, authenticate first 
-- POST localhost:5000/auth/login
-- header: application/json
-- content:

```
{
	"username": "Jacek",
	"password": "pass123"
}
```

- POST: `/participants/`
- add authorization token (from previous step)
-  content:

```
{
	"name": "wihajster",
	"lastname": "wihajster",
	"email": "wistler@wistler.pl",
	"github": "githubgithub",
	"phone": "123456"
}
```

### CODE REVIEW

**Comment by w1stler** [comment]
I won't add test as we move to rest.

File: backend/resources/participant.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 
-from flask import Blueprint, jsonify, request
+from flask import request
 from flask_jwt_extended import jwt_required
 from flask_restful import Resource
 
 from backend.models import Participant
-from backend.serializers.participant_serializer import participants_schema
+from backend.serializers.participant_serializer import participants_schema, participant_schema
+
```
Comment (stanislawK) [inline backend/resources/participant.py]: Line length > 79 characters

File: backend/resources/participant.py [hunk 2]
```diff
@@ -16,3 +20,28 @@ def get(self):
             participants = participants_schema.dump(participants_list)
             return {"participants": participants}, HTTPStatus.OK
         return {"message": "Participants not found"}, HTTPStatus.NOT_FOUND
+
+    @jwt_required
+    def post(self):
+        json_data = request.get_json(force=True)
+        if not json_data:
+            return {'message': 'No input data provided'}, HTTPStatus.BAD_REQUEST
```
Comment (stanislawK) [inline backend/resources/participant.py]: Line length > 79 characters

File: backend/resources/participant.py [hunk 3]
```diff
@@ -16,3 +20,28 @@ def get(self):
...
+        try:
+            data = participant_schema.load(json_data)
+        except ValidationError as err:
+            return (err.messages), HTTPStatus.BAD_REQUEST
+
+        participant = Participant.query.filter_by(name=data['name']).first()
+        if participant:
+            return ({'message': 'Participant already exists.'}), HTTPStatus.BAD_REQUEST
```
Comment (stanislawK) [inline backend/resources/participant.py]: Line length > 79 characters

File: backend/resources/participant.py [hunk 4]
```diff
@@ -16,3 +20,28 @@ def get(self):
...
+            name=json_data['name'],
+            lastname=json_data['lastname'],
+            email=json_data['email'],
+            github=json_data['github'],
+            phone=json_data['phone']
+        )
+        db.session.add(participant)
+        db.session.commit()
+
+        return ({"message": "Participant created successfully."}), HTTPStatus.CREATED
```
Comment (stanislawK) [inline backend/resources/participant.py]: Line length > 79 characters

**Comment by stanislawK** [comment]
Works well for me. Only line length.

File: backend/resources/participant.py [hunk 5]
```diff
@@ -1,9 +1,16 @@
+from backend.extensions import db
```
[the remark this answers is not in the document]
Comment (w1stler) [reply backend/resources/participant.py]: I think it's sorted right now

File: backend/resources/participant.py [hunk 6]
```diff
@@ -14,3 +21,32 @@ def get(self):
...
+            return {'message': 'No input data provided'}, \
+                HTTPStatus.BAD_REQUEST
+        try:
+            data = participant_schema.load(json_data)
+        except ValidationError as err:
+            return (err.messages), HTTPStatus.BAD_REQUEST
+
+        participant = Participant.query.filter_by(name=data['name']).first()
+        if participant:
+            return ({'message': 'Participant already exists.'}), \
```
[the remark this answers is not in the document]
Comment (w1stler) [reply backend/resources/participant.py]: Done.

File: backend/resources/participant.py [hunk 7]
```diff
@@ -14,3 +21,32 @@ def get(self):
...
+                HTTPStatus.BAD_REQUEST
+        participant = Participant(
```
[the remark this answers is not in the document]
Comment (w1stler) [reply backend/resources/participant.py]: Done.

File: backend/resources/participant.py [hunk 8]
```diff
@@ -14,3 +21,32 @@ def get(self):
...
+            name=data['name'],
+            lastname=data['lastname'],
+            email=data['email'],
+            github=data['github'],
+            phone=data['phone']
+        )
+        db.session.add(participant)
+        db.session.commit()
+
+        return ({"message": "Participant created successfully."}), \
```
[the remark this answers is not in the document]
Comment (w1stler) [reply backend/resources/participant.py]: Dpme.

File: backend/resources/participant.py [hunk 9]
```diff
@@ -14,3 +21,32 @@ def get(self):
...
+        json_data = request.get_json(force=True)
+        if not json_data:
+            return {'message': 'No input data provided'}, \
+                HTTPStatus.BAD_REQUEST
+        try:
+            data = participant_schema.load(json_data)
+        except ValidationError as err:
+            return (err.messages), HTTPStatus.BAD_REQUEST
+
+        participant = Participant.query.filter_by(name=data['name']).first()
```
[the remark this answers is not in the document]
Comment (w1stler) [reply backend/resources/participant.py]: Only name is set to be required right now. I would say email should be unique and validation based on that.

[the remark this answers is not in the document]
Comment (w1stler) [reply backend/resources/participant.py]: I will remove this condition, as we are going to make changes to the schema. I would suggest to go with email as unique key.

**Pull request merged**